src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1897,8 +1897,42 @@ impl Cage {
         0 //chdir has succeeded!;
     }
 
-    //------------------------------------DUP & DUP2 SYSCALLS------------------------------------
-
+    ///##------------------------------------DUP & DUP2 SYSCALLS------------------------------------
+    /// ## `dup_syscall`
+    ///
+    /// ### Description
+    /// This function duplicates a file descriptor. It creates a new file
+    /// descriptor that refers to the same open file description as the original file descriptor.
+    /// * Finding the Next Available File Descriptor: If `start_desc` is provided and it is already in use, the function 
+    ///   will continue searching for the next available file descriptor starting from `start_desc`. If no file 
+    ///   descriptors are available, it will return an error (`ENFILE`).
+    /// * If `fd` is equal to `start_fd`, the function returns `start_fd` as the new file 
```

Comment:
when both are same it means to create a fd that points to the same file.
- I think this is because the original fd already points to the desired file description.
- Also creating a new fd and adding it into fd table can add a bit of overhead, by returning we can avoid it